qa-tests-backend/src/test/groovy/ClustersTest.groovy

```diff
@@ -16,21 +16,34 @@ class ClustersTest extends BaseSpecification {
 
     def "Test cluster status has cert expiry"() {
         when:
-        "Get the cluster, and the sensor-tls cert"
+        "Get the cluster, and the sensor TLS certificates (sensor-tls and tls-cert-sensor)"
         def cluster = ClusterService.getCluster()
         assert cluster
-        def sensorTLSSecret = orchestrator.getSecret("sensor-tls", "stackrox")
-
-        then:
-        "Verify the cluster has sensor cert expiry information, and that is matches what's in the secret"
         def expiryFromCluster = new Date(
             cluster.getStatus().getCertExpiryStatus().getSensorCertExpiry().getSeconds() * 1000
         )
         assert expiryFromCluster
+
+        def sensorTLSSecret = orchestrator.getSecret("sensor-tls", "stackrox")
         def sensorCert = Cert.loadBase64EncodedCert(sensorTLSSecret.data["sensor-cert.pem"])
         def expiryFromCert = sensorCert.notAfter
         assert expiryFromCert
-        assert expiryFromCert == expiryFromCluster
+
+        def expiryFromNewCert = null
+        try {
+            // tls-cert-sensor is a new secret that may exist in the cluster due to certificate rotation.
+            def newSensorTLSSecret = orchestrator.getSecret("tls-cert-sensor", "stackrox")
+            def newSensorCert = Cert.loadBase64EncodedCert(newSensorTLSSecret.data["cert.pem"])
+            expiryFromNewCert = newSensorCert.notAfter
+        } catch (Exception e) {
+            // tls-cert-sensor doesn't exist yet, which is fine
+            log.debug("tls-cert-sensor secret not found or could not be loaded: ${e.message}")
+        }
+
+        then:
+        "Verify the cluster has sensor cert expiry information, and that is matches what's in the secret"
+        assert expiryFromCluster == expiryFromCert ||
+            (expiryFromNewCert != null && expiryFromCluster == expiryFromNewCert)
```

Comment:
This must be some groovy magic that I don't understand, but just wanted to make sure that if expiry from new cert is a pointer type (as comparison to `null` would imply), then we're not comparing pointers in the last condition, but still comparing the values.

Reply:
Yes, in Groovy `==` is shorthand for `.equals()` in Java, so it performs a value check.
But apparently the more idiomatic way to do what I was doing is the following:

```Groovy
assert expiryFromCluster == expiryFromCert || (expiryFromNewCert && expiryFromCluster == expiryFromNewCert)
```